### PR TITLE
[ONL-8065] Replace jest with vitest

### PR DESCRIPTION
**Pull request by mcibique**
- Replaced jest with vitest
- Replaced all jest mocks with vitest mocks
- Updated eslint rules to match the plugin for jest
- Removed `.toMatchDiffSnapshot` because there's no plugin like that for vitest
- Updated all snapshots because vitest serialiser uses different naming
- Removed CVueWrapper. We're now extending VueWrapper directly. This allows us to use all generics that VueWrapper provides, e.g. `.findByDataTest<HTMLAnchorElement>('my-anchor')` instead of `findByDataTest('my-anchor').element as HTMLAnchorElement`.
- Vitest coverage has a better detection than previous one so it detected tons of places in SFC templates which were not covered by tests. This lead to adding tests for these and sometimes fixing issues (e.g. data-test attrs in some components were not merged but replaced). I also rewritten tests for ec-tooltip to be able to cover all cases.
- Replaced `(aMethod as jest.Mock)` with`vi.mocked(aMethod)` for better type safety that is provided by vitest.
- Removed as many `any` as I could in the test
- Fixed typos, missing describe blocks

### CODE REVIEW

File: src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js [hunk 1]
```diff
@@ -115,7 +136,7 @@ describe('EcBtnDropdown', () => {
         },
         {
           attrs: {},
-          href: '/payments/?add-new-payment',
+          href: '#/payments/',
```
Comment (wolandec) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js:139]: Just curiosity, why `href` is changed?

File: tests/utils/global.d.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
-import { VueWrapper } from '@vue/test-utils';
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (wolandec) [inline tests/utils/global.d.ts:1]:
```suggestion
```
T on line 11 could be removed as well?

**Review by wolandec — APPROVED** [review]

**Review by derrick-g — APPROVED** [review]

File: src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js (hunk 1 above)
Comment (mcibique) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js:139]: I was getting errors from JSDOM that navigation is not implemented only hash navigation is supported. We have tests where we click on anchors with this href - to remove the errors, I changed the hrefs to hashes.

File: tests/utils/global.d.ts (hunk 2 above)
Comment (mcibique) [reply tests/utils/global.d.ts]: Yeap, removed.

**Review by wolandec — APPROVED** [review]

**Review by stavros-tomas — APPROVED** [review]

**Pull request merged**